cookbook/agent_concepts/user_control_flows/confirmation_required_stream.py

```diff
@@ -0,0 +1,70 @@
+"""🤝 Human-in-the-Loop: Adding User Confirmation to Tool Calls
+
+This example shows how to implement human-in-the-loop functionality in your Agno tools.
+It shows how to:
+- Handle user confirmation during tool execution
+- Gracefully cancel operations based on user choice
+
+Some practical applications:
+- Confirming sensitive operations before execution
+- Reviewing API calls before they're made
+- Validating data transformations
+- Approving automated actions in critical systems
+
+Run `pip install openai httpx rich agno` to install dependencies.
+"""
+
+import json
+
+import httpx
+from agno.agent import Agent
+from agno.models.openai import OpenAIChat
+from agno.tools import tool
+from agno.utils import pprint
+
+
+@tool(requires_confirmation=True)
+def get_top_hackernews_stories(num_stories: int) -> str:
+    """Fetch top stories from Hacker News.
+
+    Args:
+        num_stories (int): Number of stories to retrieve
+
+    Returns:
+        str: JSON string containing story details
+    """
+    # Fetch top story IDs
+    response = httpx.get("https://hacker-news.firebaseio.com/v0/topstories.json")
+    story_ids = response.json()
+
+    # Yield story details
+    all_stories = []
+    for story_id in story_ids[:num_stories]:
+        story_response = httpx.get(
+            f"https://hacker-news.firebaseio.com/v0/item/{story_id}.json"
+        )
+        story = story_response.json()
+        if "text" in story:
+            story.pop("text", None)
+        all_stories.append(story)
+    return json.dumps(all_stories)
+
+
+agent = Agent(
+    model=OpenAIChat(id="gpt-4o-mini"),
+    tools=[get_top_hackernews_stories],
+    markdown=True,
+)
+
+for run_response in agent.run("Fetch the top 2 hackernews stories", stream=True):
```

Comment:
On running this one-
The tool name repeats twice `get_top_hackernews_stories`. Something which can be fixed in the way we are formatting ig
<img width="1131" alt="image" src="https://github.com/user-attachments/assets/97935ac6-5799-447f-94f6-778b2adea35d" />

Reply:
Great catch. Fixed.